dsgrid/registry/dataset_registry_manager.py

```diff
@@ -129,17 +131,78 @@ def _check_lookup_data_consistency(self, config: DatasetConfig, load_data_lookup
 
     def _check_dataset_time_consistency(self, config: DatasetConfig, load_data):
         time_dim = config.get_dimension(DimensionType.TIME)
-        tz = time_dim.get_tzinfo()
+        data_tz = time_dim.get_data_tzinfo()
         time_ranges = time_dim.get_time_ranges()
+        time_ranges_tz = time_dim.get_tzinfo()
+
         assert len(time_ranges) == 1, len(time_ranges)
         time_range = time_ranges[0]
         # TODO: need to support validation of multiple time ranges: DSGRID-173
 
         expected_timestamps = time_dim.list_time_range(time_range)
-        actual_timestamps = [
-            x.timestamp.astimezone().astimezone(tz)
-            for x in load_data.select("timestamp").distinct().sort("timestamp").collect()
-        ]
+
+        #### Need to test this entire block ####
+        if data_tz != time_ranges_tz:
+            # convert data timestamps to time_ranges_tz and compare to expected_timestamps
+            if time_ranges_tz == TimeZone.LOCAL:
+                # load map as spark df and raise error if not found, make sure to_id is in proper tz_info type
+                tz_map  # FIXME #models_to_dataframe #spark.read.option("header",True).csv(filename)
+            else:
+                # create a map based on time_ranges_tz
+                tz_map = load_data_lookup.select(
+                    F.col("geography").alias("from_id"), F.lit(time_ranges_tz).alias("to_id")
+                )
+
+            timestamps_by_geography = load_data.join(
+                load_data_lookup.join(
+                    tz_map, load_data_lookup.geography == tz_map.from_id, "left"
+                ),
+                "id",
+                "left",
+            ).select(
+                "id",
+                "geography",
+                "timestamp",
+                F.lit(data_tz).alias("from_timezone"),
+                F.col("to_id").alias("to_timezone"),
+            )
+            udf_find_time_delta = F.udf(lambda fm, to: find_time_delta(fm, to), T.DateType())
```

Comment:
I don't think I'm following the type you are creating here. The find_time_delta function returns a Python timedelta object and I don't see how that gets stored in a DateType. Do you want to convert the timedelta to an integer expressed as number of seconds?

Also, note that UDFs in Python are very slow. There may not be a better way.

Reply:
You are right, I did not look super hard, maybe `CalendarIntervalType` is more appropriate? https://sparkbyexamples.com/pyspark/pyspark-sql-types-datatype-with-examples/

Reply:
I'm not sure that that is actually a Spark type. https://spark.apache.org/docs/latest/sql-ref-datatypes.html

`timedelta.total_seconds()` stored as an integer is probably good enough.